### PR TITLE
Fix: Added creation date of asset header

### PR DESCRIPTION
**Pull request by ghost**
**Description**
This PR is made to fix date issues when uploading an asset to the drive.
Before, the asset's creation date was not correct: it was the date when the asset was uploaded because no date header was provided, not when if was originally created.
Now, the date is sent and the server knows the correct date of the asset when it was created.
**How does it work ?**
The application have to send to the server the creation date. In order to do that, the application now sends in the headers the creation date of the PHAsset.
Before sending the PHAsset's creation date through headers, it formats the date correctly, conforming to the server's date format.
Then, it gathers the asset's date and adds a header with this:
`[request setValue:[df stringFromDate:[asset creationDate]] forHTTPHeaderField:@"Date"];`

### CODE REVIEW

**Comment by y-lohse** [comment]
The first 2 commits shouldn't be in this PR ;)

File: src/ios/ListLibraryItems.m [hunk 1]
```diff
@@ -171,6 +171,11 @@ - (void)uploadItem:(CDVInvokedUrlCommand *)command {
                             
                             for(NSString * header in [headers allKeys]) {
                                 [headers setObject:[self getMimeTypeFromPath:resource.originalFilename] forKey:@"Content-Type"];
+                                // Adding creation date of asset to headers
+                                NSDateFormatter * df = [[NSDateFormatter alloc] init];
```
Comment (y-lohse) [inline src/ios/ListLibraryItems.m:175]: what does `df` stand for?

Comment (ghost) [reply src/ios/ListLibraryItems.m:175]: `df` stands for DateFormatter, it formats correctly the date, matching to server's date format.

Comment (y-lohse) [reply src/ios/ListLibraryItems.m:175]: Maybe we can use a more explicit name? Like `creationDate` or something similar?

Comment (Crash--) [reply src/ios/ListLibraryItems.m:175]: In fact, we create a "dateformater" object and then we ask to this formater to... format the creationDate in the right format. So `df` is a good name I think ;). (see

``` [request setValue:[df stringFromDate:[asset creationDate]] forHTTPHeaderField:@"Date"];```
)

**Review by Crash-- — APPROVED** [review]
LGTM thanks 👍

File: src/ios/ListLibraryItems.m (hunk 1 above)
Comment (y-lohse) [reply src/ios/ListLibraryItems.m:175]: Ah I see, alright!

**Pull request merged**